core/iwasm/libraries/lib-wasi-threads/tid_allocator.h

```diff
@@ -8,8 +8,14 @@
 
 #include "platform_common.h"
 
+#ifdef __cplusplus
+extern "C" {
+#endif
```

Comment:
no, gtest is implemented fully in C++, therefore test files must be C++ files as well.

Reply:
FYI, i have been using cmocka for the other C project on macOS and ubuntu w/o any major issues.
(i'm not saying cmocka is better than other framework. actually i dunno.
i'm just saying i can share my experience.)

sample code: https://github.com/yamt/toywasm/blob/master/test/test.c

its output:
```
[==========] tests: Running 8 test(s).
[ RUN      ] test_leb128
[       OK ] test_leb128
[ RUN      ] test_endian
[       OK ] test_endian
[ RUN      ] test_functype
[       OK ] test_functype
[ RUN      ] test_idalloc
[       OK ] test_idalloc
[ RUN      ] test_timeutil
[       OK ] test_timeutil
[ RUN      ] test_timeutil_int64
[       OK ] test_timeutil_int64
[ RUN      ] test_list
[       OK ] test_list
[ RUN      ] test_xstrnstr
[       OK ] test_xstrnstr
[==========] tests: 8 test(s) run.
[  PASSED  ] 8 test(s).
```

Reply:
Thanks for sharing; I don't have a strong opinion, I used gtest because it nicely integrates with CMake, is quite widely used, well documented, and I'm quite familiar with it. But let's see if others have any preferences, I don't mind moving to other framework.

Reply:
one negative thing in cmocka is that its "mocking" functionality assumes `ld --wrap` and thus doesn't work on macOS. (i simply don't use the mocking functionality.)

Reply:
I've used gtest in the past for c code. There are some annoyances, like having to explicitly cast the return type of a malloc to compile `char* c = (char*)malloc(10);`. From what I remember it has no (great) support for mocking in c.
For the rest, I don't recall any meaningful constraints.